### PR TITLE
operator sdk update

### PR DESCRIPTION
**Pull request by goern**
- update to operator-sdk v1.22.2, relocked dependencies
- :arrow_up: update direct dependencies, relocked dependencies
- :arrow_up: update controller-gen, re-run
- check for the type of ArrayOrString, return if its a string
- remove the use of the CaseSensitiveJSONIterator

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -68,87 +69,110 @@ all: build
...
+CONTROLLER_TOOLS_VERSION ?= v0.9.2
 
-CONTROLLER_GEN = $(shell pwd)/bin/controller-gen
-controller-gen: ## Download controller-gen locally if necessary.
-	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1)
+KUSTOMIZE_INSTALL_SCRIPT ?= "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"
+.PHONY: kustomize
+kustomize: $(KUSTOMIZE) ## Download kustomize locally if necessary.
+$(KUSTOMIZE): $(LOCALBIN)
+	curl -s $(KUSTOMIZE_INSTALL_SCRIPT) | bash -s -- $(subst v,,$(KUSTOMIZE_VERSION)) $(LOCALBIN)
```
Comment (codificat) [inline Makefile:159]: Can we do something else here?

Anything other than `curl ... | bash` sounds better to me

**Review by codificat — COMMENTED** [review]
lgtm, but I would much prefer avoiding `curl | bash`

File: Makefile (hunk 1 above)
Comment (goern) [reply Makefile:159]: very true, it has been generated by the operator-sdk AFAICT

Comment (codificat) [reply Makefile:159]: oh well... I am patching that locally for the time being

**Pull request merged**